### DISCUSSION TITLE
0.5.1

### DISCUSSION THREAD

**Discussion by z-huang** [z-huang/music#926] [Announcements]
- Login support
- Mood & Genres
- Better stats screen
- Bookmark artists
- Minor enhancement and bug fixes

<hr /><em>This discussion was created from the release <a href='https://github.com/z-huang/InnerTune/releases/tag/v0.5.1'>0.5.1</a>.</em>

**Comment by mattcarter11** [comment]
Hi @z-huang, great work with this app, really!!!

One question, what does the login enable exactly? And what will allow the users to have once it's gully fletched out (if there's still things to improve/implement)

**Reply by z-huang** [reply]
Login lets you access your YouTube playlists, so you can migrate to InnerTune more easily. Also, if you have YouTube Premium, you can enjoy higher audio quality (~256 kbps) in InnerTune after logged in.

**Reply by mattcarter11** [reply]
Mm... so I'm not seeing my entire YTM playlist library. Are there any limitations on which playlist you can see? (For example, uploaded songs?)

**Comment by Rahulsourcecode** [comment]
I am facing Some issues  if i search a song and i tap on a song it starts the particular song, within 1 second it changes to some other rendom track

**Reply by nthcneuqmeuiqwr** [reply]
Yeah i have this problem to and it's not only when i search for songs

**Reply by ANDROPRROS** [reply]
The same thing happens to me, it's somewhat annoying but it only happens when you search for a song or choose one of the recommendations at the beginning

**Comment by DeadSecTr** [comment]
There are 2 Trackers in the app since last update!!! Why?! Please remove them.

**Reply by nthcneuqmeuiqwr** [reply]
I think that is because innertune now has login support. Maybe the creator could add a version without the trackers

**Reply by z-huang** [reply]
See #935.

**Comment by humood96** [comment]
Thanks for you working ! I like your app.
Just i have a question , After login i cant find section of my playlist ?

**Reply by z-huang** [reply]
<img width=200 src="https://github.com/z-huang/InnerTune/assets/39427017/6c47b9de-dd34-4cee-b523-142d899902cc" />

**Reply by humood96** [reply]
This icon is nit showen with me. Am sure i login. 
![Uploading Screenshot_20230805-094342.jpg…]()

**Comment by radical4ever** [comment]
I love it, finally i can have back my yt music olaylist, great work!!, i hope on next releases we could get some sort of widget for the home screen like the yt music's one!

**Comment by netrunner-exe** [comment]
@z-huang, why you remove a basic and very necessary feature like changing a track with a swipe?  How does this affect the stability of the application?  A lot of people have been waiting for this feature since the old version, if we talk about me, I didn’t use the old one until the new one came out with this feature.

**Reply by z-huang** [reply]
#931 and #936(probably) are caused by the horizontal pager. I couldn't find a way to solve the issues without removing the horizontal pager. To make the app more stable and usable, I had to remove the feature first before I find other ways to get around it.

**Reply by netrunner-exe** [reply]
@z-huang Okay, thanks for the reply.  In this case, I will stay on the version where the swipe is not removed from the code, since without it 90 percent of usability is lost for me - this is the most frequent feature that I use. I have one more question for you - after you have added Firebase to the repository and accordingly google-services.json to gitignore - I can no longer compile the apk on my PC in order to test the application, since this file is missing. How to be in that case?

**Reply by netrunner-exe** [reply]
@z-huang I don’t know how much the swipe affects these bugs, since I only had them in the latest releases, I decided to download for example, this build - https://github.com/z-huang/InnerTune/actions/runs/5557521364 (the swipe had already been implemented a long time ago) and compared it with the latest release.  In the debug version that I downloaded, this bug did not appear, unlike the release.  From this we can conclude that it is possible that it is not the swipe that is to blame for these bugs, but maybe something else.

**Reply by z-huang** [reply]
> without it 90 percent of usability is lost for me

Why? It's just a more convenient way to switch song. There're buttons to do it, too! Though I don't encounter the bugs, I have to take care for users who have the bugs. These bugs will greatly reduce their user experience.

As for the compiling problem, edit `app/build.gradle.kts` and remove the plugins for gms and firebase. Then you won't need `google-services.json`.

**Reply by z-huang** [reply]
> I don’t know how much the swipe affects these bugs, since I only had them in the latest releases, I decided to download for example, this build - [z-huang/InnerTune/actions/runs/5557521364](https://github.com/z-huang/InnerTune/actions/runs/5557521364) (the swipe had already been implemented a long time ago) and compared it with the latest release. In the debug version that I downloaded, this bug did not appear, unlike the release. From this we can conclude that it is possible that it is not the swipe that is to blame for these bugs, but maybe something else.

I can confirm #931 happens because of the swipe feature. But I'm not sure about #936, though it's very likely...

**Reply by netrunner-exe** [reply]
@z-huang All in all, this is very strange.  I opened the latest release version and after a couple of minutes of rearranging and playing the tracks in the queue, I managed to reproduce this bug twice.  Then, after a while, I purposefully tried to repeat it, but after 15 minutes of rearranging and playing tracks (I also periodically used a swipe to change the track and open the queue again), it did not appear.  I would also not rule out the possibility that these are some kind of bugs not related to the swipe. As for the swipe - for me this is a basic option that is in all players - Deezer, YT Music, Spotify, YT Music Revanced, etc., which I always use and have done it before.  For modern players, this can be called a standard feature.  Just because it's convenient.  I speak only for myself and for my habits and preferences.  Yes, of course, you can use the buttons, but... Personally, for me it is more convenient and natural to quickly swipe on the cover than to tap a button. Perhaps it is more convenient for someone to change tracks with buttons, but I only speak for myself.

**Comment by netrunner-exe** [comment]
> #931 and #936(probably) are caused by the horizontal pager. I couldn't find a way to solve the issues without removing the horizontal pager. To make the app more stable and usable, I had to remove the feature first before I find other ways to get around it.

@z-huang, you completely right about the swipe.  I decided to do an experiment - I compiled two versions - one with the latest commit and the second one with the latest commit but with a swipe.  The version with the swipe turned out to have these bugs ... Strange, I don’t understand why it has such an effect, because before the swipe was also there, but this problem was not observed at all, besides, I quite often updated the debug version from actions on GitHub.  At some point after the update, it began to appear periodically and I reported this by creating a topic.  I hope you manage to fix this problem and get the swipe back again!  Thank you.

**Comment by snow-viktor** [comment]
I really like it, the music playback interface is better than YouTube Music.

(I am a Traditional Chinese user) When I played some English songs, I originally wanted to sing while watching the subtitles, but the subtitles were in Chinese. Is this a bug please?

**Reply by z-huang** [reply]
At the right-bottom of the lyrics view is a button (...). You can find other lyrics there.

**Reply by snow-viktor** [reply]
Thank you. I hadn't noticed before that there was a button to search for subtitles.

**Comment by Shivansh426** [comment]
When i am searching and playing any song the song changes automatically please fix this bug

**Reply by netrunner-exe** [reply]
Are you using the latest release or debug build? If release - try to install debug, removing track switching by swiping on the album cover ([this commit](https://github.com/z-huang/InnerTune/commit/7c3bb2b349eaef4fcc31a8100a87eb7c5675dc7e)) should have solved this problem.

**Reply by Shivansh426** [reply]
Nope 👎

**Comment by adambrangenberg** [comment]
Hey, when is it going to be pushed to F Droid?

**Reply by z-huang** [reply]
This version won't be pushed to F-Droid, see #935.

**Reply by adambrangenberg** [reply]
Okay
